Feature/transaction delay

lskld opened the pull request:
### What has been done:

- Created method called ProcessPendingTransactions(); that checks all pending transactions if they were created more than 15 minutes ago. KEEP IN MIND. The method will only work if paired with a Timer that performs it/checks it (ex every minute). The method alone cannot process the transactions automatically. 

- The logic for performing the transfer of funds from account => account has been moved from the Transfer(); method to make sure no transactions go through immediately. Transfer(); method now only checks for faults and sets transaction as (PENDING).

lskld committed:
Created method for processing and timestamping transactions

lskld committed:
Minor changes and updated method name to better suit its use.

lskld committed:
Added new comments and deleted old ones

EmmaKarlholm commented:
There are a lot of warnings. Could something be done to please the compiler a bit? The functionality seems to work just as it should though!

Svyatoslav-Lapshin reviewed (changes requested):
**Question**
Is it possible just send transaction to ApplyTransaction method?
**Feedback**
Otherwise really good job

lskld committed:
Name and logic changes

lskld commented:
### Changes:

Changed to more clear method names. 
Changed the ProcessPendingTransactions method so that it only filters pending transactions, and the method will be executed every 15 minutes instead of every minute in Main method. (Timer in Main not implemented yet).

lskld committed:
Created Periodic Timer that executes ProcessPendingTransaction method every 15 minutes

lskld commented:
### Final Changes:
Created a Periodic Timer in Main method that performs the ProcessPendingTransactions(); from TransactionManager.cs every 15 minutes.
